internal/common/utils/http.go

```diff
@@ -52,15 +60,28 @@ func doGet(client HttpRequestDoer, url string, timerFactory func() *prometheus.T
 	return client.Do(req)
 }
 
-func IsGzip(reader io.Reader) (bool, error) {
-	bufferedReader := bufio.NewReaderSize(reader, 2)
+func GetCompressionType(reader io.Reader) (string, error) {
```

Comment:
I don't like how we're operating on strings rather than a more clearly defined set of values. If you make a typo in one of those strings the compiler won't be able to tell you. Using constants/enums would be better.